gemgis/vector.py

```diff
@@ -19,100 +20,522 @@
+
+    # Checking that all elements of the list are of type int or float
+    if isinstance(bbox, list) and not all(isinstance(i, (int, float)) for i in bbox):
+        raise TypeError('Bbox values must be of type float or int')
+
+    # Checking that drop_id is of type bool
+    if not isinstance(drop_id, bool):
+        raise TypeError('Drop_id argument must be of type bool')
+
+    # Checking that reset_index is of type bool
+    if not isinstance(reset_index, bool):
+        raise TypeError('Reset_index argument must be of type bool')
+
+    # Checking that the target_crs is of type string
+    if not isinstance(target_crs, (str, type(None), pyproj.crs.crs.CRS)):
+        raise TypeError('target_crs must be of type string or a pyproj object')
+
+    # Checking that overwrite_xy is of type bool
+    if not isinstance(overwrite_xy, bool):
+        raise TypeError('Overwrite_xy argument must be of type bool')
+
+    # Checking that X and Y are not in the GeoDataFrame
+    if not overwrite_xy and {'X', 'Y'}.issubset(gdf.columns):
+        raise ValueError('X and Y columns must not be present in GeoDataFrame before the extraction of coordinates')
+
+    # Copying GeoDataFrame
+    gdf = gdf.copy(deep=True)
+
+    # Reprojecting coordinates to provided target_crs
+    if target_crs is not None:
+        gdf = gdf.to_crs(target_crs)
+
+    # Extracting x,y coordinates from point vector data
+    gdf['X'] = gdf.geometry.x
+    gdf['Y'] = gdf.geometry.y
+
+    # Dropping id column
+    if 'id' in gdf and drop_id:
+        gdf = gdf.drop('id', axis=1)
+
+    # Limiting the extent of the data
+    if bbox is not None:
+        gdf = gdf[(gdf.X > bbox[0]) & (gdf.X < bbox[1]) & (gdf.Y > bbox[2]) & (gdf.Y < bbox[3])]
+
+    return gdf
+
+
+def explode_multilinestrings(gdf: gpd.geodataframe.GeoDataFrame,
+                             reset_index: bool = True,
+                             drop_level0: bool = True,
+                             drop_level1: bool = True,
+                             ) -> gpd.geodataframe.GeoDataFrame:
+    """
+    Exploding Shapely MultiLineStrings to Shapely LineStrings
+    Args:
+        gdf (gpd.geodataframe.GeoDataFrame): GeoDataFrame created from vector data containing elements of geom_type
+        MultiLineStrings
+        reset_index (bool): Variable to reset the index of the resulting GeoDataFrame, default True
+        drop_level0 (bool): Variable to drop the level_0 column, default True
+        drop_level1 (bool): Variable to drop the level_1 column, default True
+    Return:
+        gdf: GeoDataFrame containing LineStrings
+    """
+
+    # Checking that gdf is of type GepDataFrame
+    if not isinstance(gdf, gpd.geodataframe.GeoDataFrame):
+        raise TypeError('Loaded object is not a GeoDataFrame')
+
+    # Check that all entries of the gdf are of type MultiLineString or LineString
+    if not all((gdf.geom_type == 'MultiLineString') | (gdf.geom_type == 'LineString')):
```

Comment:
Would have to check if this works first, but

```suggestion
    if not all(gdf.geom_type.isin(['MultiLineString', 'LineString'])):
```